lib/mllp/sender.ex

```diff
@@ -250,15 +247,7 @@ defmodule MLLP.Sender do
         {:reply, reply, state}
 
       {:error, reason} ->
-        telemetry(
-          :status,
-          %{status: :disconnected, error: reason, context: "send message failure"},
-          state
-        )
-
-        new_state = maintain_reconnect_timer(state)
-        reply = {:error, :send_message_failure}
-        {:reply, reply, new_state}
+        handle_send_error(state, reason)
```

Comment:
I see where all the send* handle_call clauses can be collapsed into one and avoid the `handle_send_error/2` helper and `handle_recv_error/2` helpers. I would do it in another PR I think, but also not opposed to  just doing it in this PR.

Reply:
I do see each different send is doing little different things primarily to handle receive response. I like multiple smaller functions as good as they are not duplicating code

Reply:
Fair enough 👍